[choaticman] iP

choaticman opened the pull request:
# **ChoaticBot**
>"I hate perfection. To be perfect is to be unable to improve any further." -  Kurotsuchi Mayuri

ChoaticBot frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/choaticman/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:

 

- [x] Managing tasks
- [x]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

choaticman committed:
Rename bot and add welcome & bye message

choaticman committed:
Add .gitignore file

choaticman committed:
Modify .gitignore file

choaticman committed:
Add echo feature

choaticman committed:
Modify .gitignore to include more files

choaticman committed:
Update README.md to reflect new bot name

choaticman committed:
Fix missing message when adding task

choaticman committed:
Add listing and adding of tasks feature

choaticman committed:
Added Task and Tasklist class to mark/unmark tasks

choaticman committed:
Add Todo, Deadline, Event

choaticman committed:
Add TextUiTesting

choaticman committed:
Add simple error handling for todo and other texts

choaticman committed:
Add delete task feature

choaticman committed:
Delete docs directory

choaticman committed:
refactor code to remove ip folder within ip repo

choaticman committed:
Create ActionManager to handle different actions. Refactor code by adding packages. Create InputProcessor to process user inputs

choaticman committed:
Merge changes from branch-Level-7

choaticman committed:
Add saving feature

choaticman committed:
Merge saving feature from branch-Level-7

choaticman committed:
Add recognising date input for Deadlines feature

choaticman committed:
Merge changes from branch-Level-8

Fui03 commented on src/main/java/ChoaticBot.java:
```diff
@@ -0,0 +1,119 @@
+import java.util.Scanner;
+import java.io.*;
+import java.nio.file.*;
```
**Suggestion**: It might be better to avoid using the asterisk `*` in this context. Could you consider an alternative approach?

Fui03 commented on src/main/java/actions/ActionManager.java:
```diff
@@ -0,0 +1,23 @@
+            case "todo", "deadline", "event" ->
+                    new CreateTask(taskList, processedInput.getAction(), processedInput.getDetails());
+            default -> throw new UnknownActionException("OOPS!!! I'm sorry, but I don't know what that means :-(");
+        };
```
**Suggestion**: Is the indentation of the `switch` statement and `case` statements used correctly? You can refer to this [link](https://se-education.org/guides/conventions/java/intermediate.html).

Fui03 commented on src/main/java/actions/CreateTask.java:
```diff
@@ -0,0 +1,36 @@
+package actions;
+
+import tasks.*;
```
**Suggestion**: It might be better to avoid using the asterisk * in this context. Stated it above.

Fui03 commented on src/main/java/ui/Ui.java:
```diff
@@ -0,0 +1,16 @@
+
+public class Ui {
+    public static void printText(String text) {
+        String textToPrint = "____________________________________________________________\n"
```
**Suggestion**: Create a constant to store `____________________________________________________________` string.
For example:
```java
public static final String LINE_SEPARATOR = "____________________________________________________________";

Fui03 reviewed (commented):
Overall, the code adheres well to the coding standards, and your approach is clear and logical. Just a few minor adjustments are needed based on the comments I've left. Once those are addressed, it should be good to go!

choaticman committed:
Refactor code to add choaticbot package

choaticman committed:
Update text-ui-test to comply with new Date acceptance

choaticman committed:
Merge changes from add-gradle-support branch

choaticman committed:
Add gradle support

choaticman committed:
Add JUnit tests for ToDos class

choaticman committed:
Add unit tests for InputProcessor class

choaticman committed:
Merge changes from branch-A-JUnit

choaticman committed:
Change build.gradle to implement Java Archive

choaticman committed:
Merge changes from branch-A-Jar

choaticman committed:
Add JavaDoc for TaskList class

choaticman committed:
Add find feature

choaticman committed:
Merge changes from branch-A-JavaDoc

choaticman committed:
Merge changes from branch-Level-9 and resolve merge conflicts

celeschai commented on src/main/java/choaticbot/ChoaticBot.java:
```diff
@@ -0,0 +1,128 @@
+            File file = new File(DATA_FILE_PATH);
+            if (file.exists()) {
+                BufferedReader reader = new BufferedReader(new FileReader(file));
+                String line;
```
Perhaps a more specific name to this String would be clearer to the reader as to what line we are looking at?

celeschai commented on src/main/java/choaticbot/actions/Action.java:
```diff
@@ -0,0 +1,17 @@
+        this.taskList = taskList;
+    }
+
+    abstract public void execute();
```
Should this name be a more specific verb that signals what is being executed?

celeschai commented on src/main/java/choaticbot/actions/Mark.java:
```diff
@@ -0,0 +1,19 @@
+import choaticbot.tasks.TaskList;
+
+public class Mark extends Action {
+    public String details;
```
Perhaps a more specific name would better signal what details this string contains and also avoid the confusion of it being a list / array ?

celeschai commented on src/main/java/choaticbot/tasks/TaskList.java:
```diff
@@ -0,0 +1,96 @@
+import java.util.ArrayList;
+import static choaticbot.ui.Ui.printLine;
+
+/**
+ * Represents a list of tasks
+ */
```
The JavaDocs here really helps to make things clearer! Good job!

celeschai commented on src/main/java/choaticbot/tasks/TaskList.java:
```diff
@@ -0,0 +1,96 @@
+ */
+public class TaskList {
+    private ArrayList<Task> tasklist;
+    //private int numberOfTask;
```
Should this be removed?

celeschai commented on src/main/java/choaticbot/ChoaticBot.java:
```diff
@@ -0,0 +1,128 @@
+import static choaticbot.ui.Ui.printText;
+
+public class ChoaticBot {
+    public TaskList tasklist;
```
Should this be plural form instead?

celeschai reviewed (commented):
Overall I really like how you organised your code into actions, inputs, ui, tasks. It makes the inner workings of this huge code base more neat and specialised to each of its functions!

choaticman committed:
Add CheckStyle with gradle

choaticman committed:
Merge changes from branch-A-CheckStyle

choaticman committed:
Add simple GUI support

choaticman committed:
Add a simple GUI

choaticman committed:
Merge changes from branch-Level-10

choaticman committed:
Refactor code to add Storage class

choaticman committed:
Add JavaDoc to all classes

choaticman committed:
Change code to comply with checkstyle

choaticman committed:
Add assertions to code

Arguments for createAction() in ActionManager should not be null.
deadlineDetails[] in execute() of CreateTask be of size 2.
eventDetails[] in execute() of CreateTask should be of size 3.
Argument for containWord() in Task should not be null.

Java assertions have been added to ensure that these contraints
are met.

choaticman committed:
Include exception for unmarkTask() in TaskList

unmarkTask() in TaskList now throws an ChoaticBotException.
This exception allows the program to handle the case on accessing
an index that is out of bounds of the array.

This allows the chatbot to continue running instead of exiting
with an error.

choaticman committed:
Merge pull request #1 from choaticman/branch-A-CodeQuality

Improve code quality by refactoring the code

choaticman committed:
Resolve merge conflicts with master branch

Resolved merge conflicts after pulling master branch from remote.
This will allow the branch to be pushed to remote branch without conflicts.

choaticman committed:
Merge pull request #2 from choaticman/branch-A-Assertions

Include assertions into code

choaticman committed:
Add task update feature

- Introduced a new updateTask() method in TaskList that allows users
  to update existing tasks with new details
- Modified Task class and its subclasses to support updating tasks.
- Implemented input validation and added support for throwing
  WrongInputFormatException to handle cases where the user provides
  incorrect input for the commands.
- Improved error handling to prevent the program from crashing on
  invalid inputs, and provide users with the correct format to input.
- Added appropriate JavaDoc to ensure clear documentation of the new
  feature.

This update enhances tasks management by allowing users to modify
tasks details after creation without the need to delete and add tasks.

choaticman committed:
Merge changes from branch-C-Update

choaticman committed:
Improve GUI and JavaDoc to provide better documentation

choaticman committed:
Merge changes from branch-A-JavaDoc

choaticman committed:
Remove all references of Duke

choaticman committed:
Add Ui.png to docs folder

choaticman committed:
Update user guide to be more comprehensive

choaticman committed:
Update user guide to correct typos

choaticman committed:
Add more comprehensive JUnit tests

choaticman committed:
Refactor code to comply with checkstyle

choaticman committed:
Add screenshot to user guide
